contrib_bots/lib/tictactoe-bot.py

```diff
@@ -0,0 +1,329 @@
+
+from __future__ import absolute_import
+from __future__ import print_function
+import copy
+import random
+from six.moves import range
+
+initial_board = [["_", "_", "_"],
+                 ["_", "_", "_"],
+                 ["_", "_", "_"]]
+
+mode = 'r'  # default, can change for debugging to 'p'
+def output_mode(string_to_print, mode):
+    if mode == "p":
+        print(string_to_print)
+    elif mode == "r":
+        return string_to_print
+
+# -------------------------------------
+class TicTacToeGame(object):
+    smarter = True
```

Comment:
It would be more user friendly if the user didn't had to change the source code to change this setting. Have you considered using of configuration file?